src/search_index.rs

```diff
@@ -528,4 +546,110 @@ impl SearchIndex {
+        let lines: Vec<&str> = content.lines().collect();
+
+        // Find which line contains the fragment
+        let mut byte_count = 0;
+        let mut fragment_line_idx = 0;
+        for (idx, line) in lines.iter().enumerate() {
+            let line_end = byte_count + line.len();
+            if fragment_offset >= byte_count && fragment_offset < line_end {
+                fragment_line_idx = idx;
+                break;
+            }
+            byte_count = line_end + 1; // +1 for newline
+        }
+
+        // Calculate the range of lines to extract (with context)
+        let start_line = fragment_line_idx.saturating_sub(context_lines);
+        let end_line = (fragment_line_idx + context_lines + 1).min(lines.len());
+
+        // Extract the segment
+        let segment_lines = &lines[start_line..end_line];
+        let segment = segment_lines.join("\n");
+
+        // Calculate the offset adjustment for highlighting
+        let mut segment_start_offset = 0;
+        for line in &lines[0..start_line] {
+            segment_start_offset += line.len() + 1; // +1 for newline
+        }
+
+        // Highlight the segment using the original highlight ranges
+        let use_colors = atty::is(Stream::Stdout);
+        let (highlight_start, highlight_end) = if use_colors {
+            ("\x1b[1;33m", "\x1b[0m") // Bold yellow
+        } else {
+            ("", "")
+        };
+
+        let mut result = String::new();
+        let mut last_end = 0;
+
+        // Sort ranges by start position
+        let mut ranges: Vec<_> = highlighted_ranges.to_vec();
+        ranges.sort_by_key(|r| r.start);
+
+        for range in ranges {
+            let abs_start = fragment_offset + range.start;
+            let abs_end = fragment_offset + range.end;
+
+            // Skip ranges outside our segment
+            if abs_end < segment_start_offset {
+                continue;
+            }
+            if abs_start >= segment_start_offset + segment.len() {
+                break;
+            }
+
+            // Adjust the range to be relative to our segment
+            let seg_start = abs_start.saturating_sub(segment_start_offset);
+            let seg_end = if abs_end > segment_start_offset {
+                (abs_end - segment_start_offset).min(segment.len())
+            } else {
+                continue;
+            };
+
+            // Add text before highlight
+            if seg_start > last_end {
+                result.push_str(&segment[last_end..seg_start]);
+            }
+
+            // Add highlighted text
+            result.push_str(highlight_start);
+            result.push_str(&segment[seg_start..seg_end]);
+            result.push_str(highlight_end);
+
+            last_end = seg_end;
+        }
+
+        // Add remaining text after last highlight
+        if last_end < segment.len() {
+            result.push_str(&segment[last_end..]);
+        }
+
+        Ok(result)
+    }
```

Comment:
Add integration tests for this, especially showing what the inputs are (file) and what the results will look like